### ISSUE TITLE
Schema with sync_mode forces change on plan

### ISSUE THREAD

**Issue by lyschoening** [fivetran/terraform-provider-fivetran#141]
We have noticed that defining a schema where any table specifies a `sync_mode` other than `null`, terraform plan always proposes to replace the schema.

**Comment by beevital** [comment]
Hey @lyschoening ! Could you please provide an example output from terraform plan / apply commands.

**Comment by lyschoening** [comment]
Hi @beevital, sorry for the delay. Here's an extract of the plan/apply of one of our (unchanged) resources.


```
Terraform will perform the following actions:

  # module.cds.fivetran_connector_schema_config.veeva_crm_1_apac will be updated in-place
  ~ resource "fivetran_connector_schema_config" "veeva_crm_1_apac" {
        id                     = "loyalties_microorganisms"
        # (2 unchanged attributes hidden)

      - schema {
          - enabled = "true" -> null
          - name    = "salesforce" -> null

          - table {
              - enabled = "true" -> null
              - name    = "Account" -> null
            }
          - table {
              - enabled = "true" -> null
              - name    = "Approved_Document_vod__c" -> null
            }
...
          - table {
              - enabled = "true" -> null
              - name    = "Product_Metrics_vod__c" -> null
            }
    }
...

      + schema {
          + enabled = "true"
          + name    = "salesforce"

          + table {
              + enabled = "true"
              + name    = "Account"
            }
          + table {
              + enabled = "true"
              + name    = "Approved_Document_vod__c"
            }
...
          + table {
              + enabled   = "true"
              + name      = "Product_Metrics_vod__c"
              + sync_mode = "HISTORY"
            }
...
    }
}
```

```
module.cds.fivetran_connector_schema_config.veeva_crm_1_apac: Modifying... [id=loyalties_microorganisms]
module.cds.fivetran_connector_schema_config.veeva_crm_1_apac: Modifications complete after 6s [id=loyalties_microorganisms]
```

**Comment by beevital** [comment]
@lyschoening it's normal behavior. Terraform doesn't actually overwrite the whole schema. 
The issue caused by the fact that `schema` field is a collection, and if it consists of one element and there's an update inside this element -> terraform plan shows replacement for the whole element.

But the provider actually should male tiny API request that updates only target table.

Does your upstream infrastructure state aligned with desired configuration after apply? Any side-effect? 
I suspect the problem is with the way how `terraform plan` depicts changes.

**Comment by lyschoening** [comment]
@beevital We don't have an issue with the initial plan nor with the apply step, our issue is that the change to the state does not get captured properly. 

When we do another plan after applying, we see the same plan again. It's as if there is something about setting `sync_mode` anywhere in the schema that prevents the updated state from being persisted.

**Comment by beevital** [comment]
> When we do another plan after applying, we see the same plan again.

Ah, okay, now it's more clear.
Looks like upstream value not saving in state file correctly.

**Comment by michael-robbins** [comment]
Yes we are seeing this as well in our plans, where Fivetran is constantly reapplying the entire schema config where the only reported difference (between the resources being destroyed/created is the `"sync_mode" = "HISTORY"`)

**Comment by beevital** [comment]
Fix release in latest v1.0.1
https://registry.terraform.io/providers/fivetran/fivetran/1.0.1
